plugin/trino-iceberg/src/main/java/io/trino/plugin/iceberg/IcebergPageSourceProvider.java

```diff
@@ -182,6 +185,17 @@ private ConnectorPageSource createDataPageSource(
             List<IcebergColumnHandle> dataColumns,
             TupleDomain<IcebergColumnHandle> predicate)
     {
+        if (!isUseFileSizeFromMetadata(session)) {
```

Comment:
Doing this in the page source provider will make a call per split, rather than per file, though it has the benefit of being distributed. If we did this in the coordinator, we'd probably want to do it in parallel for the entire batch. And then we might want some smarter logic to list the full directory if there were more than some number of files in the same directory. So probably keep this simple, asi for now.